src/nfts/NftsInfoCarousel.tsx

```diff
@@ -0,0 +1,369 @@
+                  `${activeNft.contractAddress}-${activeNft.tokenId}` ===
+                  `${nft.contractAddress}-${nft.tokenId}`
+                }
+                nft={nft}
+              />
+            </View>
+          )
+        })}
+      </ScrollView>
+    </View>
+  )
+}
+
+type Props = NativeStackScreenProps<StackParamList, Screens.NftsInfoCarousel>
+
+export default function NftsInfoCarousel({ route }: Props) {
+  const { nfts } = route.params
+  const [activeNft, setActiveNft] = useState(nfts[0] ?? null)
+  const [isLoading, setIsLoading] = useState(true)
+  const [scaledHeight, setScaledHeight] = useState(DEFAULT_IMAGE_HEIGHT)
+  const { t } = useTranslation()
+
+  function pressExplorerLink() {
+    navigate(Screens.WebViewScreen, {
+      uri: `${networkConfig.celoExplorerBaseTokenUrl}${activeNft.contractAddress}/instance/${activeNft.tokenId}/metadata`,
+    })
+  }
+
+  function handleThumbnailPress(nft: Nft) {
+    setActiveNft(nft)
+  }
+
+  // Display error screen no Nfts provided
+  if (nfts.length === 0) {
+    return <NftsLoadErrorScreen />
+  }
+
+  return (
+    <SafeAreaView edges={['top']} style={styles.safeAreaView} testID="NftsInfoCarousel">
+      <ScrollView>
+        {/* Main Nft Image */}
+        {activeNft.metadata ? (
+          <FastImage
+            testID="NftsInfoCarousel/MainImage"
+            style={[
+              styles.mainImage,
+              { height: scaledHeight },
+              // Put a border radius on the image when loading to match placeholder
+              isLoading && styles.borderRadius,
+            ]}
+            source={{
+              uri: activeNft.media.find((media) => media.raw === activeNft.metadata?.image)
+                ?.gateway,
+            }}
+            onLoad={(e) => {
+              setScaledHeight(
+                scaleImageHeight(e.nativeEvent.width, e.nativeEvent.height, variables.width)
+              )
+            }}
+            onLoadEnd={() => setIsLoading(false)}
+            onError={() =>
+              Logger.error(
+                TAG,
+                `Error loading Nft image for Nft contractAddress: ${activeNft.contractAddress} tokenId: ${activeNft.tokenId}`
+              )
+            }
+            resizeMode={FastImage.resizeMode.contain}
+          >
+            {isLoading && <MainImagePlaceholder height={scaledHeight} />}
+          </FastImage>
+        ) : (
+          <View style={styles.nftImageLoadingErrorContainer}>
+            <Text style={{ color: colors.light }}>{t('nftInfoCarousel.nftImageLoadError')}</Text>
+          </View>
+        )}
+        {/* Display a carousel selection if multiple images */}
+        {nfts.length > 1 && (
+          <NftImageCarousel
+            nfts={nfts}
+            activeNft={activeNft}
+            handleOnPress={handleThumbnailPress}
+          />
+        )}
+        {/* Nft Details */}
+        {activeNft.metadata && (
+          <>
+            {activeNft.metadata?.name && (
```

Comment:
super nit: do you need all these chain operators? you check for metadata already, do you still need to do `metadata?.name` here (and also for other fields below?)

Reply:
There are some NFTs which do not have all metadata fields required by ERC-721. Unfortunately, [YourValoraPersona](https://explorer.celo.org/mainnet/token/0xc49F7d91b18731edC54eC5693aD43bb1dDf6553C/instance/48903/metadata) is an example that does not include a description field.